projects/plugins/backup/src/js/hooks/useConnection.js

```diff
@@ -39,7 +45,7 @@ export default function useConnection() {
 
 	const renderConnectScreen = () => {
 		return (
-			<ConnectScreen
+			<ConnectScreenWithConnectionStatus
```

Comment:
Why did we choose to use the HOC here, since we have already fetched the connection status earlier (instead of passing it as a prop)?

Reply:
The connection status we fetch from the store is empty until the `ConnectScreenWithConnectionStatus` provides that via the callback. It doesn't come from the initial state.
Once the `withConnectionStatus` fetches it via API, it's sent back to `useConnection()` via the callback, and saved into the store, so it's the other way around 🙂

Reply:
Makes sense. Many thanks for explaining, Sergey!